File: backend/src/main/java/com/app/backend/dao/entities/Reponse.java
package com.app.backend.dao.entities;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;

import java.util.Date;

@Document
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Reponse {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date dateReponse;
    private boolean visible;
    private String texte;
    @DBRef
    private User medecin;
    @DBRef
    private Question question;

}

File: frontend/src/app/shared/Reponse/reponse.ts
import { Question } from "../Question/question";
import { RegisterUser } from "../register-user";
import { User } from "../user";

export class Reponse {
  id: string;
  dateReponse: Date;
  visible: boolean;
  texte: string;
  medecin: RegisterUser;
  question: Question;

  constructor(
    id: string,
    dateReponse: Date,
    visible: boolean,
    texte: string,
    medecin: RegisterUser,
    question: Question
  ) {
    this.id = id;
    this.dateReponse = dateReponse;
    this.visible = visible;
    this.texte = texte;
    this.medecin = medecin;
    this.question = question;
  }
}
